### PR TITLE
Replace __FILE__ with __dir__ and other minor chefstyle fixes

### PR DESCRIPTION
**Pull request by tas50**
A few new chefstyle rules to improve readability here.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: spec/mixlib/log/formatter_spec.rb [hunk 1]
```diff
@@ -17,7 +17,7 @@
 #
 
 require "time" unless defined?(Time.zone_offset)
-require File.expand_path(File.join(File.dirname(__FILE__), "..", "..", "spec_helper"))
+require File.expand_path(File.join(__dir__, "..", "..", "spec_helper"))
```
Comment (phiggins) [inline spec/mixlib/log/formatter_spec.rb:20]:
```suggestion
require "spec_helper"
```

`spec/` is automatically on the load path so this is likely equivalent.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -18,7 +18,7 @@
 #
 
 $TESTING = true
-$:.push File.join(File.dirname(__FILE__), "..", "lib")
+$:.push File.join(__dir__, "..", "lib")
```
Comment (phiggins) [inline spec/spec_helper.rb:21]: `lib/` is automatically added to the load path in rspec so this is likely unnecessary.

Comment (tas50) [reply spec/spec_helper.rb:21]: @phiggins can you make that a suggestion?

Comment (phiggins) [inline spec/spec_helper.rb:21]:
```suggestion
```

**Review by phiggins — APPROVED** [review]

**Pull request merged**